src/functionality/cart.js

```diff
@@ -0,0 +1,90 @@
+const store = require('store2');
+const Orders = require('../collection/resources/orders');
+const Order = require('../collection/objects/order');
+
+class Cart {
+  constructor(request) {
+    this._cart = store.get('cart') || [];
+    this.request = request;
+  }
+
+  _findProductIndex(productId) {
+    return this._cart.findIndex(cartItem => cartItem.productId === productId);
+  }
+
+  _isCartValid() {
+    this._cart.forEach((item) => {
+      if (Object.keys(item).length !== 2 || !item.productId || item.quantity) {
+        return false;
+      }
+    });
+    return true;
+  }
+
+  _saveCart() {
+    store.set('cart', this._cart);
+  }
+
+  set(newCart) {
+    this._cart = newCart;
+  }
+
+  get() {
+    return this._cart;
+  }
+
+  empty() {
+    this._cart = [];
+  }
+
+  addProduct({ productId, quantity }) {
+    const exisitingProduct = this._findProductIndex(productId);
+    // If product item exists, we replace with new product item
+    if (exisitingProduct > -1) {
+      this._cart[exisitingProduct] = { productId, quantity };
```

Comment:
Shouldn't this add up the quantities?

Reply:
Didn't want to force any add/subtract logic, so you just replace it with the new quantity